tests/suite-unit/PluginFlyvemdmPackage.php

```diff
@@ -37,13 +37,84 @@ class PluginFlyvemdmPackage extends CommonTestCase {
 
    public function beforeTestMethod($method) {
       switch ($method) {
+         case 'testShowForm':
+         case 'testPrepareInputForAdd':
+         case 'testAddNeededInfoToInput':
          case 'testPrepareInputForUpdate':
          case 'testPostGetFromDB':
             $this->login('glpi', 'glpi');
             break;
       }
    }
 
+   /**
+    * @param $method
+    */
+   public function afterTestMethod($method) {
+      switch ($method) {
+         case 'testShowForm':
+         case 'testPrepareInputForAdd':
+         case 'testAddNeededInfoToInput':
+         case 'testPrepareInputForUpdate':
+            parent::afterTestMethod($method);
+            \Session::destroy();
+            break;
+      }
+   }
+
+   /**
+    * @return object
+    */
+   public function createInstance() {
+      $instance = $this->newTestedInstance();
+      return $instance;
+   }
```

Comment:
remove this method. it is useless to make a method for a single line of code.

Reply:
This is used on testShowForm and testAddNeededInfoToInput, to have the same instance for testing on each test

Reply:
Understood

Reply:
tests must be independent. Each test must use its own instance.

Moreover, atoum uses child processes : one child is created per test method, then it simply dies then you actually have a distinct instance for each test method.